src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -198,10 +208,25 @@ protected SCMEvent(SCMEvent<P> copy) {
      * @return a {@link ScheduledExecutorService}.
      */
     @NonNull
-    protected static ScheduledExecutorService executorService() {
-        // Future-proofing, if we find out that events are drowning Timer then we may need to move them to their
-        // own dedicated ScheduledExecutorService thread pool
-        return Timer.get();
+    protected static synchronized ScheduledExecutorService executorService() {
+        if (executorService == null) {
+            // corePoolSize is set to 10, but will only be created if needed.
+            // ScheduledThreadPoolExecutor "acts as a fixed-sized pool using corePoolSize threads"
+            executorService = new ImpersonatingScheduledExecutorService(new ScheduledThreadPoolExecutor(10, new NamingThreadFactory(new ClassLoaderSanityThreadFactory(new DaemonThreadFactory()), "SCMEvent")), ACL.SYSTEM);
+        }
+        return executorService;
+    }
+
+
+    /**
+     * Shutdown the timer and throw it away.
+     */
+    @Terminator
+    public synchronized void closeExecutorService() {
```

Comment:
Actually I didnt think about it..

And I removed static because Im not sure if `@Terminator` can work on static method..
But if works we should add that static back to be on safe side..

Reply:
I see even @stephenc mentioned it should be static..
I will create fast fixing PR..

Reply:
https://github.com/jenkinsci/scm-api-plugin/pull/68